agent/utils/utils.go

```diff
@@ -105,6 +106,31 @@ type ForwardedRequest struct {
 	Contents *http.Request
 }
 
+// CookieCache represents a LRU cache to store session ID -> CookieJar
+type CookieCache struct {
+	cache *lru.Cache
+	mu    sync.Mutex
+}
+
+// NewCookieCache initializes an LRU cache with cookieCacheLimit entries
+func NewCookieCache(cookieCacheLimit int) (*CookieCache, error) {
+	return &CookieCache{
+		cache: lru.New(cookieCacheLimit),
+	}, nil
+}
+
+// AddJarToCache takes a Jar from http.Client and stores it in a cache
+func (cj *CookieCache) AddJarToCache(sessionID string, jar http.CookieJar) {
+	cj.mu.Lock()
+	cj.cache.Add(sessionID, jar)
+	cj.mu.Unlock()
+}
+
+// GetCachedCookieJar returns the Jar mapped to the sessionID
+func (cj *CookieCache) GetCachedCookieJar(sessionID string) (interface{}, bool) {
+	return cj.cache.Get(sessionID)
```

Comment:
We should do the type assertion in this method and return an `http.CookieJar` instead of just `interface{}`

e.g.

```go
	val, ok := cj.cache.Get(sessionID)
	if !ok {
		return nil, ok
	}
	jar, ok := val.(http.CookieJar)
	return jar, ok
```